charts/ingress-nginx/templates/controller-service-internal.yaml

```diff
@@ -16,6 +16,9 @@ metadata:
   namespace: {{ include "ingress-nginx.namespace" . }}
 spec:
   type: "{{ .Values.controller.service.type }}"
+{{- if hasKey .Values.controller.service.internal "allocateLoadBalancerNodePorts" }}
+  allocateLoadBalancerNodePorts: {{ .Values.controller.service.internal.allocateLoadBalancerNodePorts }}
```

Comment:
Maybe just use a default value here instead of if/else: https://helm.sh/docs/chart_template_guide/functions_and_pipelines/#using-the-default-function

eg.: 
```
allocateLoadBalancerNodePorts:  {{ .Values.controller.service.internal.allocateLoadBalancerNodePorts | default true }}
```

Reply:
Sorry, nevermind, my approach would be a breaking change for folks that set this parameter manually :)